hips/hip-9999.md

```diff
@@ -0,0 +1,278 @@
+
+### Why `artifactType` first, then `config.mediaType` fallback?
+
+The OCI Image Specification explicitly states that "artifacts have historically been created without an `artifactType` field, and tooling to work with artifacts should fallback to the `config.mediaType` value." Following this guidance ensures compatibility with:
+
+1. Charts pushed by current Helm versions (which don't set `artifactType`)
+2. Charts pushed by third-party tools that may not set `artifactType`
+3. Registries that strip or don't preserve `artifactType`
+
+### Why `artifactType` equals `config.mediaType`
+
+The OCI Image Spec (descriptor.md) explicitly defines `artifactType` in Index descriptors as "the value of the config descriptor `mediaType` when the descriptor references an image manifest." This equality is intentional — it exposes the artifact identifier at the Index level for efficient selection without fetching manifests. Using `application/vnd.cncf.helm.config.v1+json` for both fields is correct per OCI spec, not a naming collision.
+
+### Why not use layer mediaType for selection
+
+Layer mediaTypes (e.g., `application/vnd.cncf.helm.chart.content.v1.tar+gzip`) are not exposed in Image Index descriptors — only `mediaType`, `digest`, `size`, `platform`, `artifactType`, and `annotations` are available. To check layers, Helm would need to fetch every manifest in the Index first, defeating the purpose of efficient artifact selection. The `artifactType`/`config.mediaType` approach provides artifact identification at the Index level without additional round-trips.
+
+### Why skip descriptors with `platform` field?
+
+Descriptors with a `platform` field are container images targeted at specific architectures. A Helm chart is not platform-specific and should never have a `platform` field. Skipping these descriptors avoids unnecessary manifest fetches.
+
+### Why use first match?
+
+The OCI Image Index specification states: "If multiple manifests match a client or runtime's requirements, the first matching entry SHOULD be used." This behavior is consistent with multi-architecture image selection.
+
+### Why add Referrers API support?
+
+The Referrers API enables discovering all artifacts (charts, SBOMs, signatures) associated with a specific container image. For Helm, this allows:
+
+- Finding the chart that deploys a specific image version
+- Ensuring chart and image are always used together
+- Enabling security scanning workflows that link vulnerabilities to deployment configurations
+
+## Specification
+
+This HIP introduces three related changes:
+
+### 1. Artifact Selection from Image Index
+
+When `helm pull`, `helm install`, or `helm dependency update` encounters an OCI reference that resolves to an Image Index (`application/vnd.oci.image.index.v1+json`), Helm MUST select a chart manifest using the following algorithm:
+
+1. **First pass**: Iterate through descriptors in `manifests[]` and check for `artifactType: application/vnd.cncf.helm.config.v1+json`
```

Comment:
I agree that no changes should be made to the internal structure of the Helm chart itself (necessitating the need to register additional IANA types). This would require a HIP of its own